### PR TITLE
[DESK-463] One click registration

### PR DESCRIPTION
**Pull request by JamieRuderman**
### Description

To support single click registration of the device and local services quick and easy.

### Checklist

- [x] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [ ] ~~I have prefixed this PR with `[WIP]` if it is a Work In Progress (not ready to merge)~~
- [x] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [x] I have added one or more reviewers
- [ ] ~~I have added/updated tests for any code changes~~
- [ ] ~~I have updated documentation (including README, inline comments and docs.remote.it docs)~~
- [x] This PR only contains one isolated change (bug fix, feature, etc)
- [ ] ~~Translation strings added/updated for all user-visible text~~
- [x] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [x] I have notified reviewers on Slack by @ mentioning them in the `#desktop`channel
- [x] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Test plan

1. Start up desktop with elevated permissions
2. Go to find.remote.it and find the system
3. Click connect
4. Login
5. You should land on the device setup screen
6. The hostname should be in the device name field
7. All local services should appear in a checked state
8. Click on the register button
9. Spinner should spin
10. Page should change and confirmation snackbar message should appear
11. After a minute, your device should appear in the device list

### Ticket

<https://remoteit.atlassian.net/browse/DESK-463>

### CODE REVIEW

File: frontend/src/helpers/nameHelper.ts [hunk 1]
```diff
@@ -41,3 +42,23 @@ export function renameServices(devices: IDevice[]) {
...
+
+export function osName(os?: Ios) {
+  const name = {
+    mac: 'Mac',
+    windows: 'Windows PC',
+    linux: 'Linux system',
+    rpi: 'RaspberryPi',
+  }
+  return os ? name[os] : 'system'
+}
```
Comment (danawoodman) [inline frontend/src/helpers/nameHelper.ts:64]: These would be great candidates for simple unit tests 👍

File: backend/src/cliInterface.ts [hunk 2]
```diff
@@ -22,6 +22,15 @@ class CLIInterface extends CLI {
           Tracker.event('device', 'delete', this.data.device.name)
         }
         break
+
+      case 'registration':
```
Comment (danawoodman) [inline backend/src/cliInterface.ts:26]: Should these "magic" keys instead be constants/globals stored in a central location?

File: frontend/src/components/Setup/Setup.tsx [hunk 3]
```diff
@@ -15,134 +17,156 @@ type Props = {
...
-          </section>
-        </form>
-      </Body>
+    <>
+      <Container
+        header={
+          <>
+            <Breadcrumbs />
+            {registered && (
+              <>
```
Comment (danawoodman) [inline frontend/src/components/Setup/Setup.tsx:58]: Wrapper tags seem redundant?

Comment (JamieRuderman) [reply frontend/src/components/Setup/Setup.tsx:58]: Yes there is an extra one hanging around. I'll remove.

File: backend/src/cliInterface.ts (hunk 2 above)
Comment (JamieRuderman) [reply backend/src/cliInterface.ts:26]: Yea, they are type-checked so they feel pretty safe. Could add as globals tool.

File: frontend/src/helpers/nameHelper.ts (hunk 1 above)
Comment (JamieRuderman) [reply frontend/src/helpers/nameHelper.ts:64]: Yea not a bad idea.

**Pull request merged**